tests/nodes/NodeConnectionManager.test.ts

```diff
@@ -0,0 +1,1819 @@
+    try {
+      const seedNodes: SeedNodes = {};
+      seedNodes[nodesUtils.encodeNodeId(remoteNodeId1)] = {
+        host: remoteNode1.revProxy.getIngressHost(),
+        port: remoteNode1.revProxy.getIngressPort(),
+      };
+      seedNodes[nodesUtils.encodeNodeId(remoteNodeId2)] = {
+        host: remoteNode2.revProxy.getIngressHost(),
+        port: remoteNode2.revProxy.getIngressPort(),
+      };
+      nodeConnectionManager = new NodeConnectionManager({
+        keyManager,
+        nodeGraph,
+        fwdProxy,
+        revProxy,
+        seedNodes,
+        logger: nodeConnectionManagerLogger,
+      });
+      await remoteNode1.nodeGraph.setNode(nodeId1, {
+        host: serverHost,
+        port: serverPort,
+      });
+      await remoteNode2.nodeGraph.setNode(nodeId2, {
+        host: serverHost,
+        port: serverPort,
+      });
+      await nodeConnectionManager.start();
+      await nodeConnectionManager.syncNodeGraph();
+      expect(await nodeGraph.getNode(nodeId1)).toBeDefined();
+      expect(await nodeGraph.getNode(nodeId2)).toBeDefined();
+      expect(await nodeGraph.getNode(dummyNodeId)).toBeUndefined();
+    } finally {
+      await nodeConnectionManager?.stop();
+    }
+  });
+  test('should handle an offline seed node when synchronising nodeGraph', async () => {
+    let nodeConnectionManager: NodeConnectionManager | undefined;
+    try {
+      const seedNodes: SeedNodes = {};
+      seedNodes[nodesUtils.encodeNodeId(remoteNodeId1)] = {
+        host: remoteNode1.revProxy.getIngressHost(),
+        port: remoteNode1.revProxy.getIngressPort(),
+      };
+      seedNodes[nodesUtils.encodeNodeId(remoteNodeId2)] = {
+        host: remoteNode2.revProxy.getIngressHost(),
+        port: remoteNode2.revProxy.getIngressPort(),
+      };
+      seedNodes[nodesUtils.encodeNodeId(dummyNodeId)] = {
+        host: serverHost,
+        port: serverPort,
+      };
+      // Adding information to remotes to find
+      await remoteNode1.nodeGraph.setNode(nodeId1, {
+        host: serverHost,
+        port: serverPort,
+      });
+      await remoteNode2.nodeGraph.setNode(nodeId2, {
+        host: serverHost,
+        port: serverPort,
+      });
+      nodeConnectionManagerLogger.setLevel(LogLevel.ERROR);
+      nodeConnectionManager = new NodeConnectionManager({
+        keyManager,
+        nodeGraph,
+        fwdProxy,
+        revProxy,
+        seedNodes,
+        connConnectTime: 500,
+        logger: nodeConnectionManagerLogger,
+      });
+      await nodeConnectionManager.start();
+      // This should complete without error
+      await nodeConnectionManager.syncNodeGraph();
+      // Information on remotes are found
+      expect(await nodeGraph.getNode(nodeId1)).toBeDefined();
+      expect(await nodeGraph.getNode(nodeId2)).toBeDefined();
+    } finally {
+      await nodeConnectionManager?.stop();
+    }
+  });
+
+  // Connection failure conditions
+  describe('should destroy and remove connection if...', () => {
```

Comment:
This is a nested describe. It should only be used when we have a necessary shared/expensive shared context not relevant to the other tests.

In such a case, the `describe` should have a better name.

It should be the `NodeConnectionManager test with ...`. Similar to `tests/bin/status.test.ts`. You should name what the shared context is.

Also better to use:

```
`${NodeConnectionManager.name} test`
```

As it statically ensures that we are naming what we are testing correctly.